Mathlib/Geometry/Manifold/VectorBundle/LocalFrame.lean

```diff
@@ -283,3 +301,302 @@ end
+    e.localFrame b i x = e.basisAt b hx i := by
+  simp [localFrame, hx]
+
+lemma localFrame_apply_of_notMem
+    (e : Trivialization F (TotalSpace.proj : TotalSpace F V → M)) [MemTrivializationAtlas e]
+    (b : Basis ι 𝕜 F) {i : ι} (hx : x ∉ e.baseSet) :
+    e.localFrame b i x = 0 := by
+  simp [localFrame, hx]
+
+variable [ContMDiffVectorBundle 1 F V I]
+
+open scoped Classical in
+variable (I) in
+/-- Coefficients of a section `s` of `V` w.r.t. the local frame `b.localFrame e i`.
+
+If x is outside of `e.baseSet`, this returns the junk value 0. -/
+noncomputable def localFrame_coeff
+    (e : Trivialization F (TotalSpace.proj : TotalSpace F V → M)) [MemTrivializationAtlas e]
+    (b : Basis ι 𝕜 F) (i : ι) : (Π x : M, V x) →ₗ[𝕜] M → 𝕜 :=
+  (e.localFrame_isLocalFrameOn_baseSet I 1 b).coeff i
+
+variable {e : Trivialization F (TotalSpace.proj : TotalSpace F V → M)} [MemTrivializationAtlas e]
+  {b : Basis ι 𝕜 F} {x x' : M}
+
+variable (e b) in
+@[simp]
+lemma localFrame_coeff_apply_of_notMem_baseSet (hx : x ∉ e.baseSet) (s : Π x : M, V x) (i : ι) :
+    e.localFrame_coeff I b i s x = 0 := by
+  simpa [localFrame_coeff] using
+    (e.localFrame_isLocalFrameOn_baseSet I 1 b).coeff_apply_of_notMem hx s i
+
+variable (e b) in
+@[simp]
+lemma localFrame_coeff_apply_of_mem_baseSet (hx : x ∈ e.baseSet) (s : Π x : M, V x) (i : ι) :
+    e.localFrame_coeff I b i s x = (e.basisAt b hx).repr (s x) i := by
+  have he := e.localFrame_isLocalFrameOn_baseSet I 1 b
+  have : e.basisAt b hx = he.toBasisAt hx := by ext j; simp [localFrame, hx]
+  exact this ▸ he.coeff_apply_of_mem hx s i
+
+variable {s s' : Π x : M, V x}
+
+lemma localFrame_coeff_sum_eq [Fintype ι] (hx : x' ∈ e.baseSet) :
+    s x' = (∑ i, (e.localFrame_coeff I b i s x') • e.localFrame b i x') := by
+  simp only [localFrame_coeff]
+  exact (localFrame_isLocalFrameOn_baseSet I 1 e b).coeff_sum_eq s hx
+
+variable (e b) in
+/-- A local frame locally spans the space of sections for `V`: for each local trivialisation `e`
+  of `V` around `x`, we have `s = ∑ i, (b.localFrame_coeff e i s) • b.localFrame e i` -/
+lemma localFrame_eventually_eq_sum_coeff_smul [Fintype ι]
+    (hxe : x ∈ e.baseSet) :
+    ∀ᶠ x' in 𝓝 x, s x' = ∑ i, (e.localFrame_coeff I b i s x') • e.localFrame b i x' :=
+  eventually_nhds_iff.mpr ⟨e.baseSet, fun _ ↦ e.localFrame_coeff_sum_eq, e.open_baseSet, hxe⟩
+
+variable (e b) in
+/-- The representation of `s` in a local frame at `x` only depends on `s` at `x`. -/
+lemma localFrame_coeff_congr {i : ι} (hss' : s x = s' x) :
+    e.localFrame_coeff I b i s x = e.localFrame_coeff I b i s' x := by
+  by_cases hxe : x ∈ e.baseSet
+  · simp [localFrame_coeff, hxe]
+    congr
+  · simp [localFrame_coeff, hxe]
+
+variable (e b) in
+lemma localFrame_coeff_apply_zero_at (hs : s x = 0) (i : ι) : e.localFrame_coeff I b i s x = 0 := by
+  simp only [localFrame_coeff]
+  exact (localFrame_isLocalFrameOn_baseSet I 1 e b).coeff_apply_zero_at hs i
```

Comment:
I'm dubious about this lemma. It's a trivial consequence of linearity so I feel something is wrong with the API if we need it.